server/app/services/email/graph/GraphApiEmailClient.java

```diff
@@ -0,0 +1,128 @@
+
+  private final EmailSendMetrics emailSendMetrics;
+  private final GraphApiClientInterface client;
+  private final Environment environment;
+  private final String sender;
+
+  @Inject
+  public GraphApiEmailClient(
+      Credentials credentials,
+      Config config,
+      Environment environment,
+      EmailSendMetrics emailSendMetrics) {
+    this.environment = checkNotNull(environment);
+    this.sender = checkNotNull(config).getString(AZURE_SENDER_CONF_PATH);
+    this.emailSendMetrics = emailSendMetrics;
+
+    if (environment.isProd()) {
+      client = new GraphApiClient(credentials);
+    } else {
+      client = new TestGraphApiClient();
+    }
+  }
+
+  @Override
+  public void send(String toAddress, String subject, String bodyText) {
+    send(ImmutableList.of(toAddress), subject, bodyText);
+  }
+
+  @Override
+  public void send(ImmutableList<String> toAddresses, String subject, String bodyText) {
+    if (toAddresses.isEmpty()) {
+      return;
+    }
+    Histogram.Timer timer = emailSendMetrics.getEmailExecutionTime().startTimer();
+
+    // Add some messaging to non-prod emails to make it easier to
+    // tell that it's not a prod notification.
+    if (!environment.isProd()) {
+
+      subject = String.format("[Test Message] %s", subject);
+      bodyText =
+          String.format(
+              "This email was generated from our test server.\n\n"
+                  + "If you didn't expect this message please disregard.\n\n"
+                  + "***************************************************\n\n\n"
+                  + "%s",
+              bodyText);
+    }
+
+    try {
+      Message message = new Message();
+      message.setSubject(subject);
+      ItemBody body = new ItemBody();
+      body.setContentType(BodyType.Text);
+      body.setContent(bodyText);
+      message.setBody(body);
+
+      Recipient senderRecipient = new Recipient();
+      EmailAddress senderEmail = new EmailAddress();
+      senderEmail.setAddress(sender);
+      senderRecipient.setEmailAddress(senderEmail);
+      message.setFrom(senderRecipient);
+
+      ArrayList<Recipient> toRecipients = new ArrayList<Recipient>();
+      toAddresses.forEach(
+          address -> {
+            Recipient recipient = new Recipient();
+            EmailAddress emailAddress = new EmailAddress();
+            emailAddress.setAddress(address);
+            recipient.setEmailAddress(emailAddress);
+            toRecipients.add(recipient);
+          });
+      message.setToRecipients(toRecipients);
+
+      SendMailPostRequestBody sendMailPostRequestBody = new SendMailPostRequestBody();
+      sendMailPostRequestBody.setMessage(message);
+      // This can be a configurable value if that is something the team wants.
+      sendMailPostRequestBody.setSaveToSentItems(false);
+      client.get().me().sendMail().post(sendMailPostRequestBody);
+    } catch (ApiException e) {
```

Comment:
would there be other type of exception other than APIException?

Reply:
The documentation is pretty bad, so I'm honestly not sure, but I think this is the main one. I was going off of these [docs](https://github.com/microsoftgraph/msgraph-sdk-java/blob/1e994e01ec38d2cb9ed41afa7eca698ac3d32732/docs/upgrade-to-v6.md?plain=1#L264-L271).